app.js

```diff
@@ -139,6 +176,29 @@ nextMarker = argv.marker || null;
 AWS.config.sslEnabled = argv.sslEnabled;
 var s3 = new AWS.S3();
 
+if (argv.listFile) {
+  listFile = new LineStream();
+  
+  if (argv.listFile === '-') {
```

Comment:
`-` is a common way to indicate reading from STDIN rather than a file.  If `-` is set, then we read from STDIN.  That allows us to pipe from an application getting a list (such as Solr in our case).